Fixed Single track cover reload on reload metadata from file

NikhilBisht2 opened the pull request:
Fixes inconsistent metadata and cover reload beaviour:

How to Test
1. Open Track Properties.
2. Test "Re-import Metadata from File" → Metadata should update without affecting the cover.
3. Test Right click on the cover "Reload from File/Folder" → Cover should update without affecting metadata.

Issue:
https://github.com/mixxxdj/mixxx/issues/14409

Swiftb0y reviewed (commented):
can you link to a bugreport for this? I don't understand the original issue.

NikhilBisht2 commented:
Issue:
https://github.com/mixxxdj/mixxx/issues/14409

NikhilBisht2 commented:
> can you link to a bugreport for this? I don't understand the original issue.

can you please go through the fixes

daschuer commented:
I have tested this a bit and I cannot confirm that the issue is fixed. 

The issue is about the single and multi tracks property dialog. You can open them by either select multiple or one track before open the track properties via the context menu. 

@ronso suggest not to reload the cover, when  pressing the reload track metadata button. Currently the cover is still reloaded in the multi track properties dialog.

NikhilBisht2 commented:
Thanks , I'll look into it again .

NikhilBisht2 commented:
@daschuer I made some changes , the cover art now doesn't change when metadata is reloaded . But some files still don't get a cover art assigned , but once assigned they don't change on metadata reload.

NikhilBisht2 commented:
So the only issue is that when multiple tracks are selected , the single track cover reload is fixed with this.

NikhilBisht2 commented:
Uploading simplescreenrecorder-2025-03-31_16.44.56.mp4…

NikhilBisht2 commented:
https://github.com/user-attachments/assets/ab748e9f-198c-477e-9448-a8161b1ca694

daschuer commented on src/library/dlgtrackinfo.cpp:
```diff
@@ -786,25 +786,37 @@ void DlgTrackInfo::slotImportMetadataFromFile() {
     const auto resetMissingTagMetadata = m_pUserSettings->getValue<bool>(
             mixxx::library::prefs::kResetMissingTagMetadataOnImportConfigKey);
+
+    // Pass false to skip cover loading
```
importCover is not passed, the comment is misleading.

daschuer commented on src/library/dlgtrackinfo.cpp:
```diff
@@ -786,25 +786,37 @@ void DlgTrackInfo::slotImportMetadataFromFile() {
             mixxx::library::prefs::kResetMissingTagMetadataOnImportConfigKey);
+
+    // Pass false to skip cover loading
+    const bool importCover = false;
```
This is a workaround for the confusing signature
`importTrackMetadataAndCoverImage()`  in case you don't want a cover. 

Is the a function that skips the cover part already? 
Probably not. Maybe it is better to invent a thin wrapper 
That only imports track metadata? This can be used in many other places in the code. Something like 
importTrackMetadata()

An alternative would be to rename the nullptr 
`const *QImage const pNoCoverImport = nullptr` 

I have not fully investigated it. Please check related code if these suggestion fits. 

If you prefer to keep the scpe limited just pass directly nullptr below.

daschuer commented on src/library/dlgtrackinfo.cpp:
```diff
@@ -786,25 +786,37 @@ void DlgTrackInfo::slotImportMetadataFromFile() {
-            trackMetadata.getAlbumInfo().getTitle(),
-            coverImage);
+
+    if (importCover) {  // Import cover only if requested
```
Can this ever be true?

daschuer reviewed (changes requested):
Works good now. Thank you. I have some code clean up suggestions.

NikhilBisht2 commented:
@daschuer 
I refactored slotImportMetadataFromFile() by removing the unused coverImage and wrapping the metadata import call in a lambda for clarity. I also introduced a constexpr pointer (pNoCoverImport) to explicitly skip cover import instead of passing nullptr directly.

daschuer commented:
Thank you. 

As a first-time contributor we need you to sign the [Mixxx Contributor Agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ) and comment here when you have done so. It gives us permission to distribute your contribution under the GPL v2 or later license and the Apple Mac App Store. It is also helpful for us to have contact information for contributors in case we may need it in the future.

NikhilBisht2 commented:
@daschuer 
Thanks a lot , I am really grateful for this opportunity .

daschuer reviewed (approved):
The code looks good. 

The commit history however can be improved. In this case it makes sense to squash all commits and give a bit more context about the fix as comment. 

In general we keep the history when merging, therefore evey commit shall describe what it contains in an imperative form. Reabsing under review is disencuraged. However ammending the last commit is alwasy possible under review. When you plan a rebase after review, you can push chages as "!fixup commit"
https://git-scm.com/docs/git-commit#Documentation/git-commit.txt-code--fixupamendrewordltcommitgtcode

After review like now rebasing and sqashing is allowed. Are you familar with git to do it?

NikhilBisht2 commented:
Yes , I'm familiar but still new to open source but I'll implement the required steps moving forward .

daschuer commented:
Ok, do you then want to squash this PR? 
It is: 
``` 
git rebase 4b1be30776cc2b28bd9271e0bfe67f8cb9c5a91d~1 -i
```
-> edit the Todo list to squash all commits `s` and save. 
-> edit the commit message with a bit more context
``` 
git push -f 
````

NikhilBisht2 commented:
All set

NikhilBisht2 committed:
Fixed the cover reload on single track

Fixed cover reload on reload metadata for multi track selection

Fixed the single track cover file reload

Fixed the cover reload on single track

Fixed cover reload on reload metadata for multi track selection

Fixed the single track cover file reload

Fixed the cover reload on single track metadata reload

Fixed the cover reload on single track

Fixed cover reload on reload metadata for multi track selection

Fixed the single track cover file reload

Fixed the cover reload on single track

Fixed cover reload on reload metadata for multi track selection

Fixed the single track cover file reload

Fixed the cover reload on single track metadata reload

Fixed the cover reload on metadata reload for single  track

NikhilBisht2 commented:
@daschuer this pr keeps failing this one check :
[pre-commit / Detecting code style issues (pull_request)](https://github.com/mixxxdj/mixxx/actions/runs/14292316582/job/40532660182?pr=14494)
can you help me with this?

Eve00000 commented:
The error marked is about the spacing missing on the 2nd line (coding guidelines).
The issue is due to not using pre-commit in your git.
Please take a look at the Mixxx-wiki: [Set up automatic code checking](https://github.com/mixxxdj/mixxx/wiki/Using%20Git#set-up-automatic-code-checking)

daschuer commented:
Unfortunately recommit is still failing. Can you just install it and theses issues are automatically solved. 
https://github.com/mixxxdj/mixxx/wiki/Coding-Guidelines
You may also apply the patch file you find here for download: 
https://github.com/mixxxdj/mixxx/actions/runs/14534040233?pr=14494

NikhilBisht2 commented:
Here's what I'm doing : 
1.git add . 
2.git commit -m "Fixed the cover reload on metadata reload for single  track"
`clang-format.............................................................Failed
- hook id: clang-format
- files were modified by this hook

[INFO] First pass: Reformatting added/changed lines...
[INFO] Reformatting src/library/dlgtrackinfo.cpp [783-783, 794-797, 803-806, 818-819, 824-824, 827-829, 831-832, 843-844, 849-849]...
[INFO] Second pass: Breaking long added/changed lines...`
`[WARNING] Stashed changes conflicted with hook auto-fixes... Rolling back fixes...
[INFO] Restored changes from /home/nikhil/.cache/pre-commit/patch1745555194-24425.`
3.git add .
4.git commit -m "Fixed the cover reload on metadata reload for single  track"
then it passes all the checks and then i push

daschuer commented:
Thank you @NikhilBisht2 
I have rebased this to 2.5. Waiting for CI.

NikhilBisht2 commented:
@daschuer If there's any other thing you want some work on, please let me know

daschuer commented:
Oh, thank you for the offer, yes, there is a lot more, just look into the issue tracker. It is most beneficial to do things you are benefit yourself from or where you you have particular interest. We may have a chat on zulip on that.
